superset/views/dashboard/views.py

```diff
@@ -102,6 +105,21 @@ def new(self):  # pylint: disable=no-self-use
         db.session.commit()
         return redirect(f"/superset/dashboard/{new_dashboard.id}/?edit=true")
 
+    @has_access
```

Comment:
found I had to run `superset init` after adding this decorator. Not sure if there's a way around it. This is necessary to override FAB's default list view.

Reply:
One possible way could be to override with:
```
Class Dashboard(BaseSupersetView):
     class_permission_name = "DashboardModelView"
         method_permission_name = {
        "new": "add",
        "list_all": "list",
    }
```

Note: This would deprecate the Dashboard.new permission